### ISSUE TITLE
Question about the loss definition of `GumbelQuantizer`

### ISSUE THREAD

**Issue by QiuJueqin** [sczhou/CodeFormer#284]
Hi, I looked into `GumbelQuantizer` module (although it is not used in your released arch) and noticed that the loss was defined as

https://github.com/sczhou/CodeFormer/blob/8392d0334956108ab53d9439c4b9fc9c4af0d66d/basicsr/archs/vqgan_arch.py#L109

I also found that you added a comment at the top of this snippet:  `# + kl divergence to the prior loss`.

To my understanding, the predicted code (or predicted logits, after some learnable projection `logits = self.proj(z)`, more precisely) by the encoder is expected to be **concentrating on one particular channel** instead of being uniformly distributed along channel axis, right? In other words, we expect one and only one element along the channel axis in `logits` to be close to 1, and others to be close to 0's.

If my understanding were correct, there should be a minus symbol before first `qy` in the loss function:

```
diff = self.kl_weight * torch.sum(-qy * torch.log(qy * self.codebook_size + 1e-10), dim=1).mean()
```

In this form, the predicted logits by the encoder will be pushed toward **being more individually certain** along the channel axis during the training. Without the minus symbol, however, the encoder was encouraged to output a uniformly distributed code.
